src/test/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNodeTest.java

```diff
@@ -0,0 +1,135 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import net.sf.saxon.Configuration;
+import net.sf.saxon.om.GenericTreeInfo;
+import net.sf.saxon.om.NodeInfo;
+
+public class AbstractNodeTest {
+    @Test
+    public void testCompareTo() {
+        final MockNode node1 = new MockNode(10, 1);
+        final MockNode node2 = new MockNode(8, 1);
+
+        assertEquals(2, node1.compareOrder(node2), "Invalid comparison result");
```

Comment:
Please use `assertWithMessage` from `Truth` instead.
There is a [bug](https://github.com/pmd/pmd/issues/2149) in PMD. It is already fixed, but not yet released.

Reply:
done